packages/fxa-settings/src/components/Settings/MfaGuard/index.tsx

```diff
@@ -71,26 +92,36 @@ export const MfaGuard = ({
   // Modal Setup
   useEffect(() => {
     (async () => {
-      const hasJwt = JwtTokenCache.hasToken(sessionToken, requiredScope);
-      if (hasJwt) {
-        setShowModal(false);
-        return;
-      }
-      // To avoid requesting multiple OTPs while awaiting user entry
-      if (hasSentConfirmationCode.current) {
+      // To avoid requesting multiple OTPs on mount
+      if (
+        hasSentConfirmationCode.current ||
+        JwtTokenCache.hasToken(sessionToken, requiredScope)
+      ) {
         return;
       }
       try {
         hasSentConfirmationCode.current = true;
         await authClient.mfaRequestOtp(sessionToken, requiredScope);
-        setShowModal(true);
       } catch (err) {
         hasSentConfirmationCode.current = false;
-        // TODO: FXA-12329 - There might be some errors to handle inline like rate-limiting.
-        handleError(err);
+        if (err.code === 401) {
+          handleError(err);
+          return;
+        }
+        alertBar.error(getLocalizedErrorMessage(ftlMsgResolver, err));
+        onDismiss();
```

Comment:
Can we add a Sentry.captureException(err) here? Since we don't call `handleError`, if some unexpected error occurs here then it'd fly under our radar and we wouldn't know.